src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -0,0 +1,217 @@
+                    var top_frame = pause_location["callFrames"][0];
+                    Assert.Equal("IntAdd", top_frame["functionName"].Value<string>());
+                    Assert.Contains("debugger-test.cs", top_frame["url"].Value<string>());
+
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, top_frame["functionLocation"]);
+
+                    //now check the scope
+                    var scope = top_frame["scopeChain"][0];
+                    Assert.Equal("local", scope["type"]);
+                    Assert.Equal("IntAdd", scope["name"]);
+
+                    Assert.Equal("object", scope["object"]["type"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, scope["startLocation"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 14, 4, scripts, scope["endLocation"]);
+                    return Task.CompletedTask;
+                }
+            );
+        }
+
+        [Theory]
+        [InlineData("c == 30", 3, 0)]
+        [InlineData("true", 3, 0)]
+        [InlineData("5", 3, 0)]
+        [InlineData("c < 40", 3, 0)]
+        [InlineData("c == 40", 3, 1)]
+        [InlineData("g == 40", 3, 1)]
+        [InlineData("c < 0", 3, 1)]
+        public async Task ConditionalBreakpointTrue(string condition, int offset_bp, int bp_index_expected)
+        {
+            Result [] bps = new Result[2];
+            bps[0] = await SetBreakpointInMethod("debugger-test.dll", "Math", "IntAdd", offset_bp, condition:condition);
+            bps[1] = await SetBreakpointInMethod("debugger-test.dll", "Math", "IntAdd", offset_bp + 1);
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs",
+                bps[bp_index_expected].Value["locations"][0]["lineNumber"].Value<int>(),
+                bps[bp_index_expected].Value["locations"][0]["columnNumber"].Value<int>(),
+                "IntAdd");
+        }
+
+        [Theory]
+        [InlineData("c == 15", 78, 3, 78, 11)]
+        [InlineData("c == 17", 78, 3, 79, 3)]
+        [InlineData("g == 17", 78, 3, 79, 3)]
+        [InlineData("true", 78, 3, 78, 11)]
+        [InlineData("5", 78, 3, 78, 11)]
+        public async Task JSConditionalBreakpoint(string condition, int line_bp, int column_bp, int line_expected, int column_expected)
+        {
+            var bp1_res = await SetBreakpoint("/debugger-driver.html", line_bp, column_bp, condition: condition);
+            var bp2_res = await SetBreakpoint("/debugger-driver.html", 79, 3);
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { conditional_breakpoint_test(5, 10); }, 1);",
+                "debugger-driver.html", line_expected, column_expected, "conditional_breakpoint_test");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointAttribute()
```

Comment:
Is the expression the only difference between this and `ConditionalBreakpointTrue`? If so, then we can just add the expression there.

Reply:
I did what you suggested...